inspec/a2-api-integration/controls/iam_v2.rb

```diff
@@ -427,63 +428,207 @@
   end
 
   describe "v2beta users API" do
-    before(:all) do
-       resp = automate_api_request("/apis/iam/v2beta/users",
-        http_method: 'POST',
-        request_body: {
-          id: CUSTOM_USER_ID,
-          name: "display name !#$#",
-          password: "chefautomate"
-        }.to_json
-      )
-      expect(resp.http_status).to eq 200
-    end
-
-    after(:all) do
-      resp = automate_api_request("/apis/iam/v2beta/users/#{CUSTOM_USER_ID}", http_method: 'DELETE')
-      expect(resp.http_status).to eq 200
-    end
-
-    # TODO more inspec tests coming very soon in A2-655
-
-    it "user can update their own display name" do
-      updatedName =  "i updated my own name"
-      resp = automate_api_request("/apis/iam/v2beta/self/#{CUSTOM_USER_ID}",
-        http_method: 'PUT',
-        request_body: {
-          name: updatedName
-        }.to_json
-      )
-      expect(resp.http_status).to eq 200
-      expect(resp.parsed_response_body[:user][:name]).to eq updatedName
-    end
-
-    it "user gets a 400 if their password is wrong" do
-      updatedName =  "i updated my own name"
-      resp = automate_api_request("/apis/iam/v2beta/self/#{CUSTOM_USER_ID}",
-        http_method: 'PUT',
-        request_body: {
-          name: updatedName,
-          password: "newpassword",
-          previous_password: "wrongagain"
-        }.to_json
-      )
-      expect(resp.http_status).to eq 400
+    TEST_USER = {
+        id: CUSTOM_USER_ID,
+        name: "display name !#$#",
+        password: "chefautomate"
+    }
+
+    describe "when the user is not yet created" do
+      CREATED_ID = "unique-user-id"
+
+      after(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/users/#{CREATED_ID}", http_method: 'DELETE')
+        expect(resp.http_status.to_s).to match(/200|404/)
+      end
+
+      describe "POST /iam/v2beta/users/:id" do
+        it "creates the user" do
+          createdName =  "i created my own name"
+          resp = automate_api_request("/apis/iam/v2beta/users",
+            http_method: 'POST',
+            request_body: {
+              id: CREATED_ID,
+              name: createdName,
+              password: 'something-new'
+            }.to_json
+          )
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:user][:name]).to eq createdName
+          expect(resp.parsed_response_body[:user][:id]).to eq CREATED_ID
+        end
+      end
+    end
+
+    describe "when multiple users exists" do
+      TEST_USER_ID_2 = 'inspec-user-2'
```

Comment:
I know using the constant is an established pattern, but I'm not clear on why we need it. I think a local variable would work fine.